index.bs

```diff
@@ -575,68 +575,182 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
+
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
+
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>ClipboardItems</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
+		A web author needs to create a |data| which is a [=ClipboardItems=] object in order to write content to clipboard using the {{Clipboard/write(data)}} method.
+		{{Clipboard/read()}} returns a [=Promise=] to [=ClipboardItems=] object that represents contents of clipboard.
+		A [=clipboard item=] can be read from [=ClipboardItems=].
 
+		<div id="clipboard-idl" dfn-for="Clipboard">
 		<div class="algorithm" data-algorithm="clipboard-read">
 		<h4 method for="Clipboard">read()</h4>
 		The {{Clipboard/read()}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
 				1. Let |r| be the result of running [=check clipboard read permission=] [=in parallel=]
 
-				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
+					Note: Clipboard permission is not supported on Safari. However, the read() method must be called inside
+					a user gesture event and the user must select the paste option from the native context menu that pops up
+					when read() is called from JS, otherwise, the promise will be rejected.
 
-				1. Let |data| be a copy of the [=system clipboard data=] represented as
-					a sequence of {{ClipboardItem}}s.
+				1. If |r| is not "granted", then [=a promise rejected with=] {{"NotAllowedError"}} DOMException in |realm|.
+
+				1. Let |data| be a copy of the [=system clipboard data=] represented as [=ClipboardItems=]. For the MIME types defined in the [=mandatory data types=] list, |data| contains the sanitized copy of text/html format, but image/png format has unsanitized payload to preserve meta data.
```

Comment:
I think you want to represent them as `[=clipboard items=]`, not `[=ClipboardItems=]`.

Then, you need to queue a task to get back to the main thread and create new `{{ClipboardItem}}`s from the `[=clipboard item=]`s. Something like:

1. Let |data| be a copy of the [=system clipboard data=] represented as [=clipboard items=]. For ...
2. [=Queue a global task=] on the ??? task source given |realm|'s [=realm/global object=] to perform the following steps:
   1. Let |items| be a `sequence<ClipboardItem>`, created in |realm|.
   2. For each [=clipboard item=] |underlyingItem| of |data|:
       1. Let |item| be a [=new=] {{ClipbardItem}}.
       2. Set |item|'s [=ClipboardItem/clipboard item=] to |underlyingItem|.
       3. Append |item| to |items|.
   3. [=Resolve=] |p| with |items|.

Reply:
Ack.